rust/src/util.rs

```diff
@@ -84,13 +84,80 @@ impl Display for Sizing {
 }
 
 #[derive(Clone, Eq, Debug)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize), serde(crate = "serde_crate"))]
 pub struct Variant {
     pub name: VariantName,
     pub tag: u8,
 }
 impl_strict_struct!(Variant, STRICT_TYPES_LIB; name, tag);
 
+#[cfg(feature = "serde")]
```

Comment:
Let's have a comment with explanation here:

```suggestion
#[cfg(feature = "serde")]
// The manual serde implementation is needed due to `Variant` bein used as a key in maps (like enum or union fields), and serde text implementations such as JSON can't serialize map keys if they are not strings. This solves the issue, by putting string serialization of `Variant` for human-readable serializers
```